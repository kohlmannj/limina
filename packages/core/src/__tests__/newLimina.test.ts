import { newLimina } from '..';

describe('newLimina', () => {
  it('works', () => {
    const heading /* css( */ = newLimina({
      fontSize: 32,
      lineHeight: 48,
      '@media (min-width: 720px)': {
        fontSize: 48,
      },
      '@media (min-width: 1280px)': {
        fontSize: 56,
      },
    });
    /* ) */
    expect(heading).toMatchSnapshot();
  });
});
