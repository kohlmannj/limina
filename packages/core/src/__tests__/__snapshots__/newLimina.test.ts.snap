// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`newLimina works 1`] = `
Object {
  "mediaQueries": Map {
    "@media (min-width: 0)" => Node {
      "after": "",
      "before": "",
      "nodes": Array [
        Node {
          "after": "",
          "before": "",
          "nodes": Array [
            Node {
              "after": " ",
              "before": "",
              "parent": [Circular],
              "sourceIndex": 0,
              "type": "media-type",
              "value": "@media",
            },
            Node {
              "after": "",
              "before": " ",
              "nodes": Array [
                Object {
                  "after": "",
                  "before": "",
                  "parent": [Circular],
                  "sourceIndex": 8,
                  "type": "media-feature",
                  "value": "min-width",
                },
                Object {
                  "after": " ",
                  "before": "",
                  "parent": [Circular],
                  "sourceIndex": 17,
                  "type": "colon",
                  "value": ":",
                },
                Object {
                  "after": "",
                  "before": " ",
                  "parent": [Circular],
                  "sourceIndex": 19,
                  "type": "value",
                  "value": "0",
                },
              ],
              "parent": [Circular],
              "sourceIndex": 7,
              "type": "media-feature-expression",
              "value": "(min-width: 0)",
            },
          ],
          "parent": [Circular],
          "sourceIndex": 0,
          "type": "media-query",
          "value": "@media (min-width: 0)",
        },
      ],
      "sourceIndex": 0,
      "type": "media-query-list",
      "value": "@media (min-width: 0)",
    },
    "@media (min-width: 720px)" => Node {
      "after": "",
      "before": "",
      "nodes": Array [
        Node {
          "after": "",
          "before": "",
          "nodes": Array [
            Node {
              "after": " ",
              "before": "",
              "parent": [Circular],
              "sourceIndex": 0,
              "type": "media-type",
              "value": "@media",
            },
            Node {
              "after": "",
              "before": " ",
              "nodes": Array [
                Object {
                  "after": "",
                  "before": "",
                  "parent": [Circular],
                  "sourceIndex": 8,
                  "type": "media-feature",
                  "value": "min-width",
                },
                Object {
                  "after": " ",
                  "before": "",
                  "parent": [Circular],
                  "sourceIndex": 17,
                  "type": "colon",
                  "value": ":",
                },
                Object {
                  "after": "",
                  "before": " ",
                  "parent": [Circular],
                  "sourceIndex": 19,
                  "type": "value",
                  "value": "720px",
                },
              ],
              "parent": [Circular],
              "sourceIndex": 7,
              "type": "media-feature-expression",
              "value": "(min-width: 720px)",
            },
          ],
          "parent": [Circular],
          "sourceIndex": 0,
          "type": "media-query",
          "value": "@media (min-width: 720px)",
        },
      ],
      "sourceIndex": 0,
      "type": "media-query-list",
      "value": "@media (min-width: 720px)",
    },
    "@media (min-width: 1280px)" => Node {
      "after": "",
      "before": "",
      "nodes": Array [
        Node {
          "after": "",
          "before": "",
          "nodes": Array [
            Node {
              "after": " ",
              "before": "",
              "parent": [Circular],
              "sourceIndex": 0,
              "type": "media-type",
              "value": "@media",
            },
            Node {
              "after": "",
              "before": " ",
              "nodes": Array [
                Object {
                  "after": "",
                  "before": "",
                  "parent": [Circular],
                  "sourceIndex": 8,
                  "type": "media-feature",
                  "value": "min-width",
                },
                Object {
                  "after": " ",
                  "before": "",
                  "parent": [Circular],
                  "sourceIndex": 17,
                  "type": "colon",
                  "value": ":",
                },
                Object {
                  "after": "",
                  "before": " ",
                  "parent": [Circular],
                  "sourceIndex": 19,
                  "type": "value",
                  "value": "1280px",
                },
              ],
              "parent": [Circular],
              "sourceIndex": 7,
              "type": "media-feature-expression",
              "value": "(min-width: 1280px)",
            },
          ],
          "parent": [Circular],
          "sourceIndex": 0,
          "type": "media-query",
          "value": "@media (min-width: 1280px)",
        },
      ],
      "sourceIndex": 0,
      "type": "media-query-list",
      "value": "@media (min-width: 1280px)",
    },
  },
  "stylesByMediaQueryString": Map {
    "@media (min-width: 720px)" => Object {
      "fontSize": 48,
    },
    "@media (min-width: 1280px)" => Object {
      "fontSize": 56,
    },
    "@media (min-width: 0)" => Object {
      "fontSize": 32,
      "lineHeight": 48,
    },
  },
}
`;
